lib/ipinfo-rails.rb

```diff
@@ -2,18 +2,25 @@
 
 require 'rack'
 require 'ipinfo'
+require 'ipinfo-rails/ipselector/default_ipselector'
 
 class IPinfoMiddleware
     def initialize(app, cache_options = {})
         @app = app
         @token = cache_options.fetch(:token, nil)
         @ipinfo = IPinfo.create(@token, cache_options)
         @filter = cache_options.fetch(:filter, nil)
+        @ip_selector = cache_options.fetch(:ip_selector, nil)
```

Comment:
We should just use `DefaultIPSelector` right here instead of `nil`, so that then a `nil` check isn't necessary in `call`.

Reply:
We need to create a new object of `DefaultIPSelector` or any `ip_selector` and pass `request` in the constructor. The request object is created after the `initialize` method because it requires `env` so I think we have to stick with this approach.

Reply:
But the user already is passing in the class `XForwardedIPSelector` in the examples, without any initializations.

Reply:
User just need to pass the class name we later after creating the request object creating a new object of the class passed by user. The `request` object is being created in the `call` function that's why we are doing a check there. 
```ruby 
def call(env)
        env['called'] = 'yes'
        request = Rack::Request.new(env)
        ip_selector = if @ip_selector.nil? 
                        DefaultIPSelector.new(request)
                      else
                        @ip_selector.new(request)
                      end
```

Reply:
Yes exactly, what I'm saying is instead of `nil` as the default, we can just use `DefaultIPSelector`. The code will later create a new version of that automatically. What am I misunderstanding?